### PR TITLE
Feature/ab2d 3779 enrollment updates weekly

### PR DESCRIPTION
**Pull request by sb-wnyffenegger**
**JIRA Tickets:**

[AB2D-3779](https://jira.cms.gov/browse/AB2D-3779) - Switch to Weekly Enrollment Updates
 
### What Does This PR Do?

Switch to loading enrollment weekly with a manual override to force loading enrollment of all enabled contracts.

* Pull enrollment (mapping contracts to benes) for the current month and previous two months every week
* Pull enrollment on Tuesday at Midnight EST unless overridden
* Only pull enrollment for enabled contracts not all attested contracts
* Add a manual override via the properties table in the database that allows triggering a load of all enabled contracts once before resetting.

### What Should Reviewers Watch For?

* That the manual override cannot get stuck in on mode and continuously trigger updates
* That the changes to pull only enabled contracts do just that
* That EOB jobs still cannot start when the enrollment is being updated

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: common/src/main/java/gov/cms/ab2d/common/service/PdpClientServiceImpl.java [hunk 1]
```diff
@@ -33,6 +37,14 @@
 
     private final Mapping mapping;
 
+    @Override
+    public List<Contract> getAllEnabledContracts() {
+        return pdpClientRepository.findAllByEnabledTrue().stream()
+                .filter(client -> client.getContract().getAttestedOn() != null)
+                .filter(client -> client.getRoles().stream().anyMatch(role -> role.getName().equals(SPONSOR_ROLE)))
```
Comment (sharshar-sb) [inline common/src/main/java/gov/cms/ab2d/common/service/PdpClientServiceImpl.java:44]: Stream within a stream? Maybe put the internal stream into another method like:
client -> hasRole(client, SPONSOR_ROLE)

**Review by sharshar-sb — CHANGES_REQUESTED** [review]

File: common/src/main/java/gov/cms/ab2d/common/service/PdpClientServiceImpl.java (hunk 1 above)
Comment (sb-wnyffenegger) [reply common/src/main/java/gov/cms/ab2d/common/service/PdpClientServiceImpl.java]: Unwrapped it see if you approve.

**Review by sharshar-sb — DISMISSED** [review]

**Review by sharshar-sb — DISMISSED** [review]

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java [hunk 2]
```diff
@@ -18,4 +18,6 @@
 
     // Number of days that job is running
     private final int stuckHours;
+
+    private final boolean override;
```
Comment (nflearl) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java:22]: Use an ENUM instead.  That would give multiple options for operating modes - like skip refresh of current month on job submission.

**Review by nflearl — CHANGES_REQUESTED** [review]

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java [hunk 3]
```diff
@@ -361,23 +382,24 @@ public boolean isCoverageAvailable(Job job) throws InterruptedException {
...
-                    String reason = String.format("%s-%d-%d must be updated before job %s can run",
-                            currentPeriod.getContract().getContractNumber(), currentPeriod.getYear(), currentPeriod.getMonth(),
-                            job.getJobUuid());
-                    log.info(reason);
-                    coverageService.submitSearch(currentPeriod.getId(), reason);
-                }
-
-                return false;
-            }
+            // TODO re-enable if patient endpoint times improve significantly
```
[the remark this answers is not in the document]
Comment (sb-wnyffenegger) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java:385]: @nflearl can we instead put this off to another ticket?

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java (hunk 2 above)
Comment (sb-wnyffenegger) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java:22]: @nflearl can we instead put this off to another ticket?

Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java:22]: Sure, just comment with the ticket number.

Comment (sb-wnyffenegger) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageUpdateConfig.java:22]: AB2D-3787

**Review by nflearl — APPROVED** [review]

**Review by sharshar-sb — DISMISSED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Review by nflearl — DISMISSED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**